### PR TITLE
CopyButton component [Closes #83, Closes #63]

### LINKED ISSUE #63: recipient address doesn't copy or go to etherscan
image

the buttons are dummy right now

### LINKED ISSUE #83: Consolidate use of copyAddress
We use this basic pattern in a bunch of components:

```
  async copyAddress(): Promise<void> {
    if (!this.hash) { return }
    try {
      await navigator.clipboard.writeText(this.hash)
      this.isCopied = true
      await new Promise(resolve => setTimeout(resolve, 1000))
      this.isCopied = false
    } catch (error) {
      console.warn('Error in copying text: ', error) /* eslint-disable-line no-console */
    }
  }
```

### PR DESCRIPTION
**Pull request by wackerow**
- Extracts `copyAddress()` logic (#83) from it's use in multiple other components into a single component called `CopyButton`
- This component takes a few props: 
  - `text` is the text to copy
  - `type` is an optional string that fills in "Copy ____" in the tooltip
  - `position` is optional for tooltip (default bottom)
  - `myClass` gives the ability to style individual uses differently 
  - `callback` allows passing a function to update the parent component when copy button has been pressed
- Added to `IpfsCopyWidget`, `TransactionReceipt`, `ProjectProfile` and `Profile`
- Removed instances of `copyAddress` no longer being used
- Fixes link to Etherscan for project recipient address (#63)

### CODE REVIEW

**Comment by wackerow** [comment]
image
Tried our `$clr-green-gradient` color for the hover background color of these buttons on ProjectProfile, curious others thoughts on how this looks.

File: vue-app/src/components/CopyButton.vue [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  isCopied = false
+
+  async copyToClipboard (): Promise<void> {
+    this.isLoading = true
+    try {
+      await navigator.clipboard.writeText(this.text)
+      this.isLoading = false
+      this.isCopied = true
+      if (typeof this.callback !== 'undefined') {
+        this.callback(true)
```
Comment (pettinarip) [inline vue-app/src/components/CopyButton.vue]: As you are not requiring this `callback` prop, I encourage you to take a look at https://vuejs.org/v2/guide/components-custom-events.html

You could do something like this:
```
this.$emit('copied', true || false)

// then from the consumer
<copy-button v-on:copied="handleCopied"></copy-button>

handleCopied(isCopied: boolean) {
  ...
}
```

If you prefer using a function prop, I would suggest having a more explicit name for the callback. Something like `onCopied`.

File: vue-app/src/components/CopyButton.vue [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+<script lang="ts">
+import Vue from 'vue'
+import Component from 'vue-class-component'
+import { Prop } from 'vue-property-decorator'
+
+import Tooltip from '@/components/Tooltip.vue'
+
+@Component({ components: { Tooltip } })
+export default class CopyButton extends Vue {  
+  @Prop() text!: string // Required: Text to copy
```
Comment (pettinarip) [inline vue-app/src/components/CopyButton.vue]: I would use `value` as the name of this prop for consistency with HTML elements.

File: vue-app/src/components/CopyButton.vue [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  @Prop() type!: string // Optional: Fills in "Copy ____" in tooltip
```
Comment (pettinarip) [inline vue-app/src/components/CopyButton.vue]: I would use `text` as the name of this prop.

**Review by pettinarip — COMMENTED** [review]
Nice job on the refactor. I left a couple of suggestions.

**Comment by pettinarip** [comment]
> Tried our `$clr-green-gradient` color for the hover background color of these buttons on ProjectProfile, curious others thoughts on how this looks.

I like the contrast between gradients and flat elements (especially having gradients on the bg). Having two different gradients in the same place...I don't like it that much.

**Comment by wackerow** [comment]
@pettinarip Thanks for the suggestions, I agree on all accounts. Pushed the updates, let me know if you have any other thoughts 👍🏻

**Review by pettinarip — APPROVED** [review]
Great!

File: vue-app/src/components/IpfsCopyWidget.vue [hunk 4]
```diff
@@ -111,20 +109,6 @@ export default class IpfsCopyWidget extends Vue {
...
-  height: 1rem;
-  padding: 0.25rem;
-  cursor: pointer;
-  &:hover {
-    background: $bg-light-color;
-    border-radius: 16px;
-  }
-}
-
-.hash-text {
```
Comment (corwintines) [inline vue-app/src/components/IpfsCopyWidget.vue:124]: I noticed through testing we lost this styling functionality where the ellipsis show: 
Screen Shot 2021-07-09 at 8 25 22 PM

I think we should try and get this back in with this PR

Comment (corwintines) [inline vue-app/src/components/IpfsCopyWidget.vue:124]: I noticed through testing we lost this styling functionality where the ellipsis show: 
Screen Shot 2021-07-09 at 8 25 22 PM

I think we should try and get this back in with this PR. I played around with adding this .hash-text class styles back, and adding that div above back in for it and seemed to solve this issue. Let me know what you think @wackerow

**Review by corwintines — CHANGES_REQUESTED** [review]
Overall good, just one request change around some lost styling in component.

**Comment by tyntyn1** [comment]
[]() @tyntyn1 @wackerow

File: vue-app/src/components/IpfsCopyWidget.vue (hunk 4 above)
Comment (wackerow) [reply vue-app/src/components/IpfsCopyWidget.vue:124]: Should be fixed with 1a9ce6e (commit name should be restores* ellipsis)

**Review by pettinarip — APPROVED** [review]

**Pull request merged**